### PR TITLE
[Feature][Connector-v2-File] Support xml file format type (#5407)

### PR DESCRIPTION
**Pull request by Ivan-gfan**
## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [x] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [x] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-connectors-v2/connector-file/connector-file-sftp/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sftp/source/SftpFileSource.java [hunk 1]
```diff
@@ -72,7 +74,7 @@ public void prepare(Config pluginConfig) throws PrepareFailException {
         if (fileFormat == FileFormat.ORC || fileFormat == FileFormat.PARQUET) {
             throw new FileConnectorException(
                     CommonErrorCode.ILLEGAL_ARGUMENT,
-                    "Sftp file source connector only support read [text, csv, json] files");
+                    "Sftp file source connector does not support read [orc, parquet] files");
```
Comment (FlechazoW) [inline seatunnel-connectors-v2/connector-file/connector-file-sftp/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sftp/source/SftpFileSource.java]: Why change this message?

**Comment by FlechazoW** [comment]
Please add e2e test case and unit test.

File: seatunnel-connectors-v2/connector-file/connector-file-sftp/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sftp/source/SftpFileSource.java (hunk 1 above)
Comment (Ivan-gfan) [reply seatunnel-connectors-v2/connector-file/connector-file-sftp/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sftp/source/SftpFileSource.java]: I thought it was more in line with the intent of the code cause I added the `xml` file type, so I changed here, should I change back? Just like `Sftp file source connector only support read [text, csv, json, (xml)] files);`

**Comment by Ivan-gfan** [comment]
> Please add e2e test case and unit test.

Got it!

File: seatunnel-connectors-v2/connector-file/connector-file-sftp/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sftp/source/SftpFileSource.java (hunk 1 above)
Comment (FlechazoW) [reply seatunnel-connectors-v2/connector-file/connector-file-sftp/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sftp/source/SftpFileSource.java]: It was unnecessary change, I thought it's better to change back. And if you want to keep it in line, why not use a static constant like 'static List<String> SUPPORT_TYPE= Lists.asList("text", "csv", "json", "xml")'?

Comment (Ivan-gfan) [reply seatunnel-connectors-v2/connector-file/connector-file-sftp/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sftp/source/SftpFileSource.java]: Okay, I'll change it back

**Comment by liugddx** [comment]
Please resolve the conflict

**Pull request closed without merging**